static/app/views/dashboardsV2/widgetBuilder/buildSteps/sortByStep/index.tsx

```diff
@@ -61,6 +63,14 @@ export function SortByStep({
   const orderBy = queries[0].orderby;
   const maxLimit = getResultsLimit(queries.length, queries[0].aggregates.length);
 
+  // We want to convert widgets to using functions in their field format (i.e. not alias form)
+  // for ordering. This check will skip the alias conversion unless the orderby is
+  // saved previously in the alias format
+  const isUsingFieldFormat =
+    trimStart(orderBy, '-') === '' ||
+    queries[0].columns.includes(trimStart(orderBy, '-')) ||
+    getAggregateAlias(trimStart(orderBy, '-')) !== trimStart(orderBy, '-');
```

Comment:
hmm, also question: why can't we just always use the function field format? ie `isUsingFieldFormat` always be true.

Reply:
> why can't we just always use the function field format? ie isUsingFieldFormat always be true

We can't use set it to true because if a saved widget is already sorting by, e.g. `count_unique_user`, then we need the values for the sort by select field to be in alias form or else it won't match and the field will look blank.

The plan is to merge this PR which figures out what values to use for the select field, then migrate all of the existing widgets to field format, and widgets going forward will be using the field format. We can eventually clean up this code that bridges the two states.

> Would it be possible to use isAggregateFieldOrEquation? (the isNumericMetrics part of isAggregateFieldOrEquation might make this a no, not sure)

I just tested this and yes I think we can use it! Thanks for pointing it out. As for `isNumericMetrics`, this applies to the release health data set and it looks like they're already saving as field format so I think the change is aligned with that. While testing that I found a small gap so I added a check for `isEquationAlias`, because if we're sorting by `equation[0]`, if a user selects a different ordering option, it should be in field format.

Reply:
Ah I see, makes sense!

> I just tested this and yes I think we can use it! Thanks for pointing it out. As for isNumericMetrics, this applies to the release health data set and it looks like they're already saving as field format so I think the change is aligned with that. While testing that I found a small gap so I added a check for isEquationAlias, because if we're sorting by equation[0], if a user selects a different ordering option, it should be in field format.

Yeah I noticed this too! Was wondering if `isUsingFieldFormat` always set to `true` would fix this. Glad you saw this and fixed though.